### PR TITLE
Added logic to produce tighter ranges for queries against geowave ind…

### PR DESCRIPTION
**Pull request by jwomeara**
…exed terms.

### CODE REVIEW

File: warehouse/query-core/src/main/java/datawave/query/util/GeoWaveUtils.java [hunk 1]
```diff
@@ -0,0 +1,860 @@
...
+                // @formatter:off
+                geometries.add(gf.toGeometry(
+                        new Envelope(
+                                bounds.getMinValuesPerDimension()[0],
+                                bounds.getMaxValuesPerDimension()[0],
+                                bounds.getMinValuesPerDimension()[1],
+                                bounds.getMaxValuesPerDimension()[1])));
+                // @formatter:on
+            }
+        }
```
Comment (keith-ratcliffe) [inline warehouse/query-core/src/main/java/datawave/query/util/GeoWaveUtils.java:740]: Wondering if all the instances of this pattern could be consolidated into one or two helper methods, similar to your `positionToGeometry` function

Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/util/GeoWaveUtils.java]: I think the latest commit has this covered.

File: warehouse/query-core/src/main/java/datawave/query/util/GeoWaveUtils.java [hunk 2]
```diff
@@ -0,0 +1,860 @@
...
+    /**
+     * Optimizes the list of byte array ranges needed to query the desired area. Portions of each range which do not intersect the original query polygon will
+     * be pruned out.
+     * 
+     * @param queryGeometry
+     *            the original query geometry used to create the list of byte array ranges
+     * @param byteArrayRange
+     *            a byte array range representing a portion of the query geometry
+     * @param rangeSplitThreshold
+     *            used to determine the minimum number of segments to break a range into - higher values will take longer to computer, but will yield tighter
```
[the remark this answers is not in the document]
Comment (jwomeara) [reply warehouse/query-core/src/main/java/datawave/query/util/GeoWaveUtils.java]: I think I'm computering the right spelling now.

**Comment by jwomeara** [comment]
I rebased with 2.9 and resolved some merge conflicts.

**Pull request merged**